core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -789,7 +789,7 @@ private void appendFloatColumn(Record record, int columnIndex) {
         final float floatValue = record.getFloat(columnIndex);
         if (floatValue == floatValue) {
             final long a = responseUtf8Sink.skip();
-            responseUtf8Sink.put(floatValue, 3);
+            responseUtf8Sink.put(floatValue);
```

Comment:
This will use `put(double)` which involves `float`->`double` conversion overhead. We should probably use `cairo.sql.float.cast.scale` here and `cairo.sql.double.cast.scale` in the `appendDoubleColumn` method. WDYT?

Reply:
The PR also adds `CharSink::put(float)` and `Numbers::append(CharSink, float)`, i do not think it should convert to double.

Reply:
Re using `cairo.sql.float.cast.scale` and `cairo.sql.double.cast.scale`, I thought that was for cast as the name suggest, and it makes sense to me to return max precision to the client.
They can cut/round the decimals as they wish on the client side.
It makes sense to set a server wide precision for the web console, so the view is consistent, but for pgwire in my view it is not the best.
Different precision could be required for different float columns.

Reply:
Yes, you're right. Those settings serve a different purpose and it indeed makes sense to use the max scales in PGWire's text format.